File: smart-english-backend/src/main/java/com/smartenglishbackend/esentity/Word.java
package com.smartenglishbackend.esentity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.elasticsearch.annotations.Document;

import java.util.List;

@Data
@Document(indexName = "words")
public class Word {
    @Id
    private String id;
    private String word;
    private String phonetic;
    private String pronunciation;
    private List<String> explanations;
    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class RelatedWords{
        private List<String> a;
        private List<String> v;
        private List<String> n;
    }
    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Example{
        private String english;
        private String chinese;
        private String audio;
    }
    private RelatedWords synonyms;
    private RelatedWords antonyms;
    private List<Example> examples;
}

File: Mobile/app/src/main/java/site/smartenglish/remote/data/Word.kt
package site.smartenglish.remote.data

typealias SearchWordResponse = List<SearchWordResponseElement>

data class GetWordResponse(
    val word: String? = null,
    val phonetic: String? = null,
    val pronunciation: String? = null,
    val explanations: List<String>? = null,
    val synonyms: SynonymsOrAntonyms? = null,
    val antonyms: SynonymsOrAntonyms? = null,
    val examples: List<Example?>? = null
)

data class SearchWordResponseElement(
    val word: String? = null,
    val explanations: List<String?>? = null
)

